services/sso/src/main/java/software.amazon.awssdk.services.sso/internal/SsoCredentialsUtils.java

```diff
@@ -0,0 +1,165 @@
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.sso.internal;
+
+import static software.amazon.awssdk.profiles.ProfileProperty.SSO_ACCOUNT_ID;
+import static software.amazon.awssdk.profiles.ProfileProperty.SSO_REGION;
+import static software.amazon.awssdk.profiles.ProfileProperty.SSO_ROLE_NAME;
+import static software.amazon.awssdk.profiles.ProfileProperty.SSO_START_URL;
+import static software.amazon.awssdk.utils.JavaSystemSetting.USER_NAME;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+import java.util.Map;
+import javax.xml.bind.DatatypeConverter;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.profiles.Profile;
+import software.amazon.awssdk.services.sso.SsoClient;
+import software.amazon.awssdk.services.sso.auth.AwsSsoCredentials;
+import software.amazon.awssdk.services.sso.model.GetRoleCredentialsRequest;
+import software.amazon.awssdk.services.sso.model.GetRoleCredentialsResponse;
+import software.amazon.awssdk.services.sso.model.RoleCredentials;
+import software.amazon.awssdk.utils.Validate;
+import software.amazon.awssdk.utils.internal.SystemSettingUtils;
+
+@SdkProtectedApi
+public final class SsoCredentialsUtils {
+
+    private static final String TOKEN_DIRECTORY = "/Users/" + SystemSettingUtils.resolveSetting(USER_NAME).get() +
+                                                  "/.aws/sso/cache/";
+
+    private Profile profile;
+    private SsoClient ssoClient;
+
+    private String ssoRegion;
+    private String ssoAccountId;
+    private String ssoRoleName;
+    private String ssoStartUrl;
+
+    private SsoCredentialsTokenProperties tokenProperties;
+
+    public SsoCredentialsUtils(Profile profile, SsoClient ssoClient) {
+        this.profile = profile;
+        this.ssoClient = ssoClient;
+    }
+
+    /**
+     * Validate and resolve the SSO profile. Then load the cached token file to retrieve necessary information for the SSO call.
+     */
+    public AwsCredentials getUpdatedCredentials() {
+        validateAndResolveSsoProfile(profile);
+
+        try {
+            String cachedTokenPath = generateCachedTokenPath(ssoStartUrl);
+            File cachedTokenFile = new File(cachedTokenPath);
+
+            tokenProperties = new ObjectMapper().readValue(cachedTokenFile,
+                                                           SsoCredentialsTokenProperties.class);
```

Comment:
NEVER use ObjectMapper directly! We don't want our libraries to have a hard-dependency on Jackson-databind. Use `JacksonUtils.sensitiveJsonNodeOf` here: (1) so that we only indirectly depend on Jackson (marginally better), and (2) so that we don't log the user's access token if parsing fails.